Assets/Scripts/KeyLocationController.cs

```diff
@@ -65,4 +84,31 @@ private IEnumerator GetPlayers()
             players.AddRange(GameObject.FindGameObjectsWithTag("Player"));
         }
     }
+
+    public void LoseHealth() //Run this when enemies are killed and reach location
+    {
+        health -= 1;
+    }
+
+    private void DestroyLocation()
+    {
+        sphere.transform.localScale -= new Vector3(shrinkValue, shrinkValue, shrinkValue) * Time.deltaTime;
+
+        foreach (GameObject player in players)
+        {
+            player.GetComponent<FirstPersonController>().keyLocationSpeedMod = 1; //reset speedmod in case a player should be slowed by the edge when the location is disabled.
+            player.GetComponent<FirstPersonController>().isPlayerInKeyLocZone = true;
+        }
+
+        if (sphere.transform.localScale.x <= 0)
+        {
+            isDestroyed = true;
+            sphere.SetActive(false);
```

Comment:
I think you should use `Destroy(gameObject)` here? If you need the position of it, just make a `Vector3` variable before destroying

Reply:
I'm wrong again :D It's a child object only, so it's fine